Source/Client.swift

```diff
@@ -126,6 +128,9 @@ extension Client {
                 }
 
                 return response
+            } else if data.count == 0 {
+                //no data received, connection got broken
+                throw ClientError.brokenConnection
```

Comment:
Does but eventually hangs, after some number of successful connections are done. If you look at the code, when the pipe gets broken, it hangs instead of throwing, so this sort of fixes it, but it's not ideal. But it's a short term fix (can be fixed properly later).